src/components/group-devices.tsx

```diff
@@ -27,51 +30,57 @@ interface GroupProps extends
 }
 
 function mapStateToProps(state: AppState, ownProps: GroupProps) {
-  const group = getOrganization(state, ownProps.navigation.state.params.groupId);
-  const devices = getDevicesWithData(state, ownProps.navigation.state.params.groupId);
-    return {
-      group,
-      devices,
-    };
+  const { groupId } = ownProps.navigation.state.params;
```

Comment:
components/devices and components/group-devices are almost identical and should be combined into a single component. One way to do this:

- If groupId is passed as a prop:
```javascript
const devices = getDevicesWithData(state, groupId);
const groups = getDescendants(state, groupId);
```
- If groupId is not passed as a prop:
```javascript
const devices = getDevicesWithData(state).filter((d) => !d.device.organizationId);
const groups = getTopLevelOrganizations(state);
```

additionally i believe add device and add group should be present at every level